lib/HeroBonus.cpp

```diff
@@ -850,6 +850,14 @@ void CBonusSystemNode::removeBonus(const std::shared_ptr<Bonus>& b)
 	CBonusSystemNode::treeHasChanged();
 }
 
+void CBonusSystemNode::removeBonuses(const CSelector &selector)
+{
+	BonusList toRemove;
+	exportedBonuses.getBonuses(toRemove, selector);

```

Comment:
Do not forget about second selector, Set it to `Selector::all` if unsure. It you omit it, function will only return `b->effectRange == Bonus::NO_LIMIT`

Reply:
Fixed. Though seems counter-intuitive that this is default behaviour.

Reply:
This is kind of legacy, good solution is to completely remove this function. (There is much to clean up in bonus API)